.github/workflows/main.yml

```diff
@@ -20,14 +20,26 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
```

Comment:
I believe this `os` line can be removed because basically the `include` is overwriting this. Just to confirm, you always want to run `ubuntu-latest` with `-pt`, right? Not two runs, one with `-pt` and one without? If so I think this `os` line can go and we just manage the options in the `include`. For reference: https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#expanding-or-adding-matrix-configurations.